### PR TITLE
=sbt Do retry connection in Player.

### PR DESCRIPTION
**Pull request by He-Pin**
refs: https://github.com/apache/incubator-pekko/issues/547
and 
refs: https://github.com/apache/incubator-pekko-http/pull/297

### CODE REVIEW

File: multi-node-testkit/src/main/scala/org/apache/pekko/remote/testconductor/Player.scala [hunk 1]
```diff
@@ -338,21 +340,36 @@ private[pekko] class PlayerHandler(
...
         reconnects -= 1
-        scheduler.scheduleOnce(nextAttempt.timeLeft)(connectionRef.set(reconnect()))
       case e => fsm ! ConnectionFailure(e.getMessage)
     }
   }
 
+  private def tryConnectToController(): Unit = {
+    Try(reconnect()) match {
+      case Success(r) => connectionRef.set(r)
+      case Failure(ex) =>
```
Comment (mdedetrich) [inline multi-node-testkit/src/main/scala/org/apache/pekko/remote/testconductor/Player.scala:362]: Is it worth it to use exponential backoff here or is that excessive?

Comment (He-Pin) [reply multi-node-testkit/src/main/scala/org/apache/pekko/remote/testconductor/Player.scala:362]: `reconnect-backoff = 1s` which is used in the `reconnect()`

**Comment by He-Pin** [comment]
```scala
[info] [JVM-2-MultiDcLastNodeMultiJvmNode2] [ERROR] [08/08/2023 10:05:01.021] [MultiDcLastNode-pekko.actor.default-dispatcher-6] [PlayerHandler(pekko://MultiDcLastNode)] Error when try to connect to remote addr:[localhost/127.0.0.1:4711] will retry, time left:[-4301592 nanoseconds], cause:[Connection refused: localhost/127.0.0.1:4711].
[info] [JVM-2-MultiDcLastNodeMultiJvmNode2] [ERROR] [08/08/2023 10:05:01.048] [MultiDcLastNode-pekko.actor.default-dispatcher-18] [PlayerHandler(pekko://MultiDcLastNode)] Error when try to connect to remote addr:[localhost/127.0.0.1:4711] will retry, time left:[980216623 nanoseconds], cause:[Connection refused: localhost/127.0.0.1:4711].
```

I tested it on my box, works as expected.

**Review by mdedetrich — APPROVED** [review]
lgtm

**Pull request merged**